test/Test_ZProc.ml

```diff
@@ -33,8 +34,30 @@ let simplification ~(zpath : string) () =
     let res = simplify z3 "(and (or (>= x 1) (> x 5)) (<= x 9))"
     in Alcotest.(check string) "same" res "(and (<= x 9) (>= x 1))")
 
+let z3_result_to_model_as_array () =
+  let result = ["sat"; "true";
+  "(model 
+                                             (define-fun x () (Array Int Int)
+                                                (_ as-array k!10!12))
+                                              (define-fun y () Int
+                                                10)
+                                                (define-fun k!10!12 ((x!0 Int)) Int
+                                                (ite (= x!0 2) 2
+                                                (ite (= x!0 1) 1
+                                                  3)))
+                                            )"]
+  in let res = match (z3_result_to_model result) with 
+                | Some model -> (let (name, res) = (List.nth_exn model 1); 
+                                  in (print_string (Value.to_string res)); let res_x = (String.equal (Value.to_string res) "(store (store ((as const (Array Int Int)) 3) 2 2) 1 1)")
+                                  in let (name, res) = (List.nth_exn model 0)
+                                  in (print_string (Value.to_string res)); res_x && (String.equal (Value.to_string res) "10")
+                                )
```

Comment:
You are assuming that the 0th element of `model` represents "y" and 1st element represents "x". That's not very robust -- it might break if we parse models in a slightly different way.

Since `model` is an association list, why not simply use `find` for "x" and "y"?

Reply:
yea, thats a better way!